lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -1915,6 +1916,11 @@ void GrandCentralPass::runOnOperation() {
     llvm::dbgs() << "\n";
   });
 
+  // per-YAML output list of interfaces.
+  SmallDenseMap<StringAttr, SmallVector<sv::InterfaceOp, 2>> interfaceYAMLMap;
```

Comment:
This `interfaceYAMLMap` is later iterated over. While this won't matter for file creation later, it can create non-deterministic IR creation (e.g., breaking tests or requiring use of `CHECK-DAG` to workaround it). Is there a data structure that will provide a stable iteration order that can be used?

Reply:
Oh, great call! Yeah let's make this deterministic, thanks!